src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -228,6 +233,37 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
             return response;
         }
 
+        internal static async Task<FunctionMetadataResponse> FunctionsMetadataRequestHandlerAsync(FunctionsMetadataRequest request)
+        {
+            var response = new FunctionMetadataResponse
+            {
+                Result = StatusResult.Success,
+                UseDefaultMetadataIndexing = false
+            };
+
+            try
+            {
+                var functionReader = new FunctionMetadataJsonReader(request.FunctionAppDirectory);
+
+                var functionMetadataList = await functionReader.ReadMetadataAsync();
```

Comment:
To improve startup perf, there's likely something else we can do here:
- Have an `IFunctionMetadataProvider.ReadMetadataAsync()`, as @kshyju suggested.
- Instead of calling it here -- call it in `StartAsync()` and store the `Task<>` (i.e. don't `await` it). 
- In this method, you can `await` that `Task<>` that you called earlier.

We know we're going to need to load that metadata, so there's no reason to wait until the host asks for it; we can start that work sooner and do it in parallel.